transaction.js

```diff
@@ -0,0 +1,146 @@
+// usersCollection.js prior to running this script. 
+
+async function main() {
+    /**
+     * Connection URI. Update <username>, <password>, and <your-cluster-url> to reflect your cluster.
+     * See http://bit.ly/NodeDocs_lauren for more details
+     */
+    const uri = "mongodb+srv://<username>:<password>@<your-cluster-url>/test?retryWrites=true&w=majority";
+
+    /**
+     * The Mongo Client you will use to interact with your database
+     * See bit.ly/Node_MongoClient for more details
+     */
+    const client = new MongoClient(uri);
+
+    try {
+        // Connect to the MongoDB cluster
+        await client.connect();
+
+        // Make the appropriate DB calls
+
+        await createReservation(client,
+            "leslie@example.com",
+            "Infinite Views",
+            [new Date("2019-12-31"), new Date("2020-01-01")],
+            { pricePerNight: 180, specialRequests: "Late checkout", breakfastIncluded: true });
+
+    } finally {
+        // Close the connection to the MongoDB cluster
+        await client.close();
+    }
+}
+
+main().catch(console.error);
+
+/**
+ * Create a reservation by storing information in both the users collection and the listingsAndReviews collection
+ * Note: this function assumes there is only one Airbnb listing in the collection with the given name.  If more than
+ * listing exists with the given name, a reservation will be created for the first listing the database finds.
+ * @param {MongoClient} client A MongoClient that is connected to a cluster with the sample_airbnb database
+ * @param {String} userEmail The email address of the user who is creating the reservation
+ * @param {String} nameOfListing The name of the Airbnb listing to be reserved
+ * @param {Array.Date} reservationDates An array of the date(s) for the reservation
+ * @param {Object} reservationDetails An object containing additional reservation details that need to be stored with the reservation
+ */
+async function createReservation(client, userEmail, nameOfListing, reservationDates, reservationDetails) {
+
+    /**
+     * The users collection in the sample_airbnb database
+     */
+    const usersCollection = client.db("sample_airbnb").collection("users");
+
+    /**
+     * The listingsAndReviews collection in the sample_airbnb database
+     */
+    const listingsAndReviewsCollection = client.db("sample_airbnb").collection("listingsAndReviews");
+
+    /**
+     * The reservation document that will be added to the users collection document for this user
+     */
+    const reservation = createReservationDocument(nameOfListing, reservationDates, reservationDetails);
+
+    // Step 1: Start a Client Session
+    // See http://bit.ly/Node_startSession for the startSession() docs
+    const session = client.startSession();
+
+    // Step 2: Optional. Define options to use for the transaction
+    const transactionOptions = {
+        readPreference: 'primary',
```

Comment:
FYI default readPreference is `primary`, but maybe not bad to be explicit here

Reply:
I pulled these options from the sample code in the docs: https://docs.mongodb.com/manual/core/transactions/.  I figured I'd leave them here for consistency.  Do you know if there are plans to change what is in the docs?

Reply:
Nope, this is fine